llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -2881,6 +2881,14 @@ bool RISCVAsmParser::parseInstruction(ParseInstructionInfo &Info,
     }
   }
 
+  // Apply mnemonic aliases because the destination mnemonic may have require
```

Comment:
Have you looked at whether we can switch the order in the generic generated code?  (non blocking, just a cleanup item)

Reply:
There are two generated functions involved here, `MatchOperandParserImpl` and `MatchInstructionImpl`. They are called separately by RISCVAsmParser.cpp. I could add a a call to applyMnemonicAlias inside `MatchOperandParserImpl`, but I wouldn't be able to remove the call inside ``MatchInstructionImpl`